lib/P2p/NetNode.cpp

```diff
@@ -776,13 +776,29 @@ bool NodeServer::handshake(CryptoNote::LevinProtocol &proto,
         return false;
     }
 
-  if(!m_node_version.empty()) {
-    if (rsp.node_data.node_version != m_node_version) {
-      logger(Logging::ERROR) << context << "COMMAND_HANDSHAKE: invoked, but peer is not running the exclusive version specified, dropping connection!";
-      return false;
+    if(!m_node_version.empty()) {
+        if (rsp.node_data.node_version != m_node_version) {
```

Comment:
This will fail 100% of the time.

Reply:
Hm, you mean if someone uses your previous exclusive version tag?

Reply:
I mean that `rsp.node_data.vesion` is different from `rsp.node_data.node_version`

Reply:
Yes, but I have not changed your function. I chaged `node_data.version = CryptoNote :: P2P_CURRENT_VERSION;` . You do not use these, if I am not mistaken. 

`node_data.node_version = m_node_version;`

I will test it now with exclusive-version tag

Reply:
In theory, our function should work independently of each other. But maybe I'm wrong, hm. I will test it now

Reply:
I might be talking out of my ass too :) Haha, I haven't tested it. I can check as well, with the changes I suggested below.  I am glad they are intended to work independently. I was probably misunderstanding your objective.

Reply:
Now I tested your function with qwertycoind --exclusive-version 5.1.4 and some nodes are blocked. I think everything should work as planned :-D I understand your objective but I was a bit confused, because I did not change anything in your function, apart from the formatting. It could have been that my change has changed something for your function.

So everything should work. But as you know as a developer, never trust your compiler 100% 😆😆

Reply:
Apparently never trust my first read-through 100% either :stuck_out_tongue: